### ISSUE TITLE
On the preprocessing of the notes

### ISSUE THREAD

**Issue by MostHumble** [JoakimEdin/medical-coding-reproducibility#14]
Hi, love what you're doing here.

wanted to ask two questions:

1- Is there a particular reason for choosing this specific preprocessing?

https://github.com/JoakimEdin/medical-coding-reproducibility/blob/24e615b10e84ece56dd71f8be4a0811c924761ff/prepare_data/prepare_mimiciv.py#L78-L88

2- I also wanted to ask about the stratification you used, I'm working on sequential disease prediction task ( predict next visit diags from previous visits ). so the targets are the next visits ICDs, and I wanted to have your thoughts on whether the stratification you used could be adapated to what I'm doing.

**Comment by JoakimEdin** [comment]
1. We chose this pre-processing because it is what previous studies used. In the paper I'm working on, I'm not doing any pre-processing except for truncating the documents to a maximum length. I found this to work well for PLM-ICD. I chose the minimum target count of 10 because I could then create a split where all the codes were in both the training and test sets. I do not think this is necessary, though, as long as you are calculating the F1 macro score as described in the paper.
2. I think I would just create random splits. Stratification is nice if you want to ensure a similar class distribution in your training, validation, and test set. I recently changed my mind and think it isn't as important as I originally thought. If you don't want classes that never occurred in your training set to impact your evaluation, you can simply ignore those classes when calculating your metrics. Do you have any specific reasons for why you want the same class distributions? If yes, I'm sure we can figure out a way to adapt the method.

**Comment by MostHumble** [comment]
Thanks a lot for the quick reponse!

1. Understood, I'll take that into consideration.
2. It is for the same reasons you've stated. The issue with simply omitting those is that we're not getting a true representation of the model's capabilities. ( we're already doing that by setting a threshold of mimimum 10 counts). 
In my task, I should perhaps give even more consideration to this, because we cannot simply ignore possible outcomes of next visits because they are rare.

Thanks again!

**Comment by JoakimEdin** [comment]
No problem! Looking forward to see your research!
